src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts

```diff
@@ -2183,6 +2186,25 @@ export async function createLayer(layerObject: any, wrap?: boolean | null, viewI
 
             newLayer = wcsLayer;
             break;
+        case 'imagery':
+            newLayer = new ImageryLayer({
+                url: layerObject.url
+            });
+
+            let imageryLayer = newLayer as ImageryLayer;
+            //if (hasValue(layerObject.renderer)) {
+            //    imageryLayer.renderer = buildJsRasterStretchRenderer(layerObject.renderer) as RasterStretchRenderer;
+            //}
+            copyValuesIfExists(layerObject, 'bandIds', 'blendMode', 'capabilities', 'compressionQuality', 'compressionTolerance',
+                'copyright', 'customParameters', 'definitionExpression', 'effect', 'fields', 'fieldsIndex', 'format',
+                'hasMultidimensions', 'imageMaxHeight', 'imageMaxWidth', 'interpolation', 'legendEnabled', 'maxScale', 'minScale',
+                'mosaicRule', 'multidimensionalInfo', 'multidimensionsionalSubset', 'noData', 'noDataInterpretation', 'objectIdField',
+                'pixelFilter', 'pixelType', 'popupEnabled', 'popupTemplate', 'rasterFields', 'rasterFunction', 'rasterFunctionInfos',
+                'refreshInterval', 'renderingRule', 'serviceRasterInfo', 'sourceJSON', 'spatialReference', 'timeExtent', 'timeInfo',
+                'timeOffset', 'useViewTime', 'version')
+
```

Comment:
This giant list is suspicious. There are a lot of complex types that need pulled out into builders, such as `popupTemplate`, `timeExtent`, `spatialREference`, etc. Only simple types (string, number, boolean) are going to work correctly in a direct copy.

Reply:
The imagery renderers have various properties that they reference in the imagerylayer.  I think this can be pared down further after testing with the other 5 renderers in GB Pro.

Reply:
My point in the first comment here was that properties with complex types should never be added to `copyValuesIfExists` in the first place, because it will silently fail. I wasn't suggesting removing _support_ for a property, but to move them to their own copying lines like 
```javascript
if (hasValue(layerObject.popupTemplate)) {
    imageryLayer.popupTemplate = buildJsPopupTemplate(layerObject.popupTemplate);
}
```
this will work, but `copyValues...` does not do any of the careful transfer that `buildJs...` does. So _all_ properties that are not `string`, `number`/`double`/`int`, `bool`, or possibly an array of one of those types, should be removed from this list _and_ implemented separately, usually with a `buildJs...` method. If you think there are properties you want to exclude from the current implementation, then they should be removed from both C# and Typescript completely, and probably added to a separate gissue so we don't forget about them.

Reply:
I did break down this list of properties.  I did identify 18 could be built out in a future gissue which creates a number of these builders that can be used by a number of different layers, etc in the future to add depth to the data sets.  As of now, I think it that many just have not been built out because the different layer data that some of these things require.  Further testing with the other imagry renderers in pro may shine some more light on if there are copy values that are failing to work?

Reply:
So, this seems to still need work. For each value in the list:
- if it works, leave it. If it is a simple type as described above, you can assume it works and leave it.
- If it doesn't work, remove it, then
  - Start a new gissue for "Adding Property support to ImageryLayer"
  OR
  - Implement copying the value with an `if (hasValue())` pattern and potentially a `buildJs...` call

Before we merge this branch, every property should be either added to a separate gissue, or implemented in a functioning copy.
Just looking at the https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-ImageryLayer.html docs, the following properties probably don't belong in this `copyValues` and need to be dealt with:
- capabilities
- customParameters
- effect
- fields
- fieldsIndex
- mosaicRule
- multidimensionalInfo
- multidimensiohalSubset
- rasterFunction
- rasterFunctionInfos
- serviceRasterInfo
- sourceJSON
- timeExtent
- timeInfo
- timeOffset

Also, there are several properties that are not in this list _and_ not implemented separately, but are important
- renderer
- popupTemplate
- portalItem
- spatialReference

Let me know if you get stuck on implementing any of these.

Reply:
In the above list the gissues were created for:
rasterFunction, MosaicRule, PixelFilter, SpatialReference, TimeInfo, RasterInfo+RasterBandInfo (is the type of serviceRasterInfo), and SourceJSON
Properties included in this PR: multidimensionalInfo, multidimensionalSubset + sub properties

Reply:
If these properties are not currently supported because they have more work to be done, they should not be in the C# class `ImageryLayer` on this commit, or should at least be commented out. Currently, you still have it in a state where to the consumer, these look like they should work, but maybe don't?

Reply:
Remaining ImageryLayer Class properties were commented out pending completion of the gissues associated with the ImageryLayer and/or other gissues that would enable the use of these particular properties.  The Time related properties have only been partially completed.